tcpclient/connection.go

```diff
@@ -1,73 +1,67 @@
 package tcpclient
 
-import "fmt"
+import (
+	"fmt"
+	"time"
+)
 
 type Connection struct {
-	ID     int
-	status connectionStatus
+	ID      int
+	status  ConnectionStatus
+	metrics connectionMetrics
 }
 
-type connectionStatus int
+type ConnectionStatus int
+
+type connectionMetrics struct {
+	tcpEstablishedDuration time.Duration
```

Comment:
This is the only point I'm not 100% convinced about. One connection can have one duration or the other; if its a failed connection, duration means time-to-file. If its a successful connection, time to dial. Do we actually need to differentiate them then? An struct may have sense in any case, as tomorrow we may want to track extra times. Maybe we have a potential interface in front of us? So we can have different types of connectionMetrics, depending on the "path" the connection took, but they can have something in common... In any case, this second part could be out of this PR scope.

Reply:
I agree, the metrics part should be extended in the future...